test/validator.test.ts

```diff
@@ -0,0 +1,371 @@
+/* eslint-disable @typescript-eslint/no-non-null-assertion */
```

Comment:
Just the tests for the core validator API. You'll notice that I opted _not_ to include unit tests for the individual API functions. I _had_ those tests in there, but felt it was overkill given the 100% test coverage we get with this higher-level set of tests. But I could probably be convinced to test the lower level APIs as well if we wanted it.